packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -532,29 +605,63 @@ const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = isolatedFie
 			if (over.fieldChange !== undefined) {
 				const wasEmpty = over.fieldChange.newContent === undefined;
 
-				// We don't have to rebase the child changes, since the other child changes don't apply to the same node
+				// TODO: Handle rebasing child changes.
 				return {
 					...change,
 					fieldChange: { ...change.fieldChange, wasEmpty },
 				};
 			}
 
-			return change;
+			const rebasedChange = { ...change };
+			const overChildChange =
+				change.deletedBy === over.deletedBy ? over.childChange : undefined;
+			const rebasedChildChange = rebaseChild(change.childChange, overChildChange);
+			if (rebasedChildChange !== undefined) {
+				rebasedChange.childChange = rebasedChildChange;
+			} else {
+				delete rebasedChange.childChange;
+			}
+
+			return rebasedChange;
 		}
 
 		if (change.childChange !== undefined) {
 			if (over.fieldChange !== undefined) {
-				// The node the child changes applied to no longer exists so we drop the changes.
-				// TODO: Represent muted changes
-				return {};
+				if (change.deletedBy === undefined) {
+					// `change.childChange` refers to the node being deleted by `over`.
+					return {
+						childChange: rebaseChild(change.childChange, over.childChange),
+						deletedBy: overTagged.revision,
+					};
+				} else if (
+					over.fieldChange.newContent !== undefined &&
+					"revert" in over.fieldChange.newContent &&
+					over.fieldChange.newContent.revision === change.deletedBy
```

Comment:
Nit: consider encpasulating those conditionals in a small helper with a nice name

Reply:
I'd like to move this out to a helper, but we'd have to do a bunch of coercing typescript into believing things aren't undefined etc. Looked at it for a minute and decided it wasn't worth it. I can add a comment here that encapsulates what we're trying to accomplish in this block if you think it'll help code clarity.

Reply:
Ah, that's right I forgot about that. A comment sounds good.